filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -1708,6 +1720,44 @@ void OpenGLDriver::makeCurrent(Handle<HwSwapChain> schDraw, Handle<HwSwapChain>
 // Updating driver objects
 // ------------------------------------------------------------------------------------------------
 
+void OpenGLDriver::setExternalIndexBuffer(Handle<HwIndexBuffer> ibh, void* externalBuffer) {
+#ifdef GL_EXT_external_buffer
+    DEBUG_MARKER()
+
+    auto& gl = mContext;
+    GLIndexBuffer* ib = handle_cast<GLIndexBuffer*>(ibh);
+
+    assert_invariant(ib->gl.isExternal);
+
+    gl.bindVertexArray(nullptr);
+    gl.bindBuffer(GL_ELEMENT_ARRAY_BUFFER, ib->gl.id);
+    glBufferStorageExternalEXT(GL_ELEMENT_ARRAY_BUFFER, 0, ib->count * ib->elementSize, externalBuffer, 0);
+
+    CHECK_GL_ERROR(utils::slog.e)
+#else
+    assert(false && "You need GL_EXT_external_buffer for setting external index buffers!");
+#endif
+}
+
+void OpenGLDriver::setExternalBuffer(Handle<HwBufferObject> boh, void* externalBuffer) {
+#ifdef GL_EXT_external_buffer
+    DEBUG_MARKER()
+
+    auto& gl = mContext;
+    GLBufferObject* bo = handle_cast<GLBufferObject*>(boh);
+
+    assert_invariant(bo->gl.isExternal);
+
+    gl.bindVertexArray(nullptr);
+    gl.bindBuffer(GL_ARRAY_BUFFER, bo->gl.id);
+    glBufferStorageExternalEXT(GL_ARRAY_BUFFER, 0, bo->byteCount, externalBuffer, 0);
+
+    CHECK_GL_ERROR(utils::slog.e)
```

Comment:
I think a runtime check for `GL_EXT_external_buffer` extension support would be appropriate here, (by surrounding this block with an `if (gl.ext.EXT_external_buffer) { ... }`). The existence of symbols of the extension in the header don't imply runtime support for it.

Reply:
Is this how you meant it? 1cfc93402afe62f4381268048b8fc2d9dc3f8aed

Reply:
You'll need to amend `struct ext` first here: https://github.com/shapr3d/filament/blob/ba804444b80d8957e005d90615983129f4501d0f/filament/backend/src/opengl/OpenGLContext.h#L131-L134
Then initialize it by checking support at startup:
https://github.com/shapr3d/filament/blob/ba804444b80d8957e005d90615983129f4501d0f/filament/backend/src/opengl/OpenGLContext.cpp#L246-L249

Reply:
(It's a nit, but it'd flip the condition and keep the happy path inside the if block, rather than having to duplicate the assert message)

Reply:
I reverted it altogether, mainly because this branch doesn't build on Windows yet, and I don't want to add changed which I cannot test. Also because these changes were cherry picked from our proto branch and later we will cherry pick again from that branch (the UWP part). Not changing the file may prevent some later conflicts. But the suggestion makes sense, so here's what I did:
 - I fixed ANGLE to report  `GL_EXT_external_buffer` as supported (https://github.com/shapr3d/angle/commit/438fa10cf87dda5565e26dad39abf13c29141f0f)
  -I modified our proto branch to check for support in runtime (https://github.com/shapr3d/filament/commit/81562b08629a700335cbe6fb476c41b28ce46681).